app/src/main/java/com/example/healthinspector/Fragments/ScanFragment.java

```diff
@@ -24,57 +30,66 @@
 public class ScanFragment extends Fragment {
 
     private FragmentScanBinding binding;
-    private CodeScanner mCodeScanner;
+    private CodeScanner codeScannerView;
+    @Override
+    public void onStart() {
+        super.onStart();
+    }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
+
+
         // Inflate the layout for this fragment
         binding = FragmentScanBinding.inflate(inflater, container, false);
         View view = binding.getRoot();
-
-        //initialize CodeScanner view
         final Activity activity = getActivity();
         CodeScannerView scannerView = binding.scannerView;
-        mCodeScanner = new CodeScanner(activity, scannerView);
-        mCodeScanner.setDecodeCallback(new DecodeCallback() {
+        codeScannerView = new CodeScanner(getActivity(), scannerView);
+
+        //resets the scannerView to look for a new barcode
+        scannerView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                codeScannerView.startPreview();
+            }
+        });
+        //if the camera permissions were granted, check the CodeScanner view for a barcode
+        //initialize CodeScanner view
+        codeScannerView.setDecodeCallback(new DecodeCallback() {
             @Override
             public void onDecoded(@NonNull final Result result) {
                 activity.runOnUiThread(new Runnable() {
                     @Override
                     public void run() {
                         //barcode is stored in result
                         //making request to OpenFoodFacts API here
-                        Toast.makeText(activity, result.getText(), Toast.LENGTH_SHORT).show();
+                        Toast.makeText(getContext(), result.getText(), Toast.LENGTH_SHORT).show();
```

Comment:
is there a `requireContext()` function? I recommend to avoid null pointer exceptions